mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -341,11 +341,11 @@ class LogisticRegression @Since("1.2.0") (
                 regParamL1
               } else {
                 // If `standardization` is false, we still standardize the data
-                // to improve the rate of convergence; as a result, we have to
-                // perform this reverse standardization by penalizing each component
-                // differently to get effectively the same objective function when
+                // to improve the rate of convergence unless the standard deviation is zero;
+                // as a result, we have to perform this reverse standardization by penalizing
+                // each component differently to get effectively the same objective function when
                 // the training dataset is not standardized.
-                if (featuresStd(index) != 0.0) regParamL1 / featuresStd(index) else 0.0
+                if (featuresStd(index) != 0.0) regParamL1 / featuresStd(index) else regParamL1
```

Comment:
Can you give an example that using `value` will fail to converge? 

I agree any non-zero number here can make the algorithm work, but should we select a particular number as the denominator, or let it be the original value? @mengxr what do you think?